src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsMatch.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+
+#pragma warning disable SA1649 // File name should match first type name
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    // IsMatch(text:s, format:s)
+    // Checks if the input text is of the correct format.
+    internal sealed class IsMatchFunction : BuiltinFunction
+    {
+        public override bool UseParentScopeForArgumentSuggestions => true;
+
+        public override bool IsSelfContained => true;
+
+        public override bool HasPreciseErrors => true;
+
+        public override bool SupportsParamCoercion => true;
+
+        public IsMatchFunction()
+            : base("IsMatch", TexlStrings.AboutIsMatch, FunctionCategories.Text, DType.Boolean, 0, 2, 3, DType.String, DType.String, DType.String)
+        {
+        }
+
+        public override IEnumerable<string> GetRequiredEnumNames()
+        {
+            return new List<string>() { LanguageConstants.MatchEnumString, LanguageConstants.MatchOptionsEnumString };
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.IsMatchArg1, TexlStrings.IsMatchArg2 };
+            yield return new[] { TexlStrings.IsMatchArg1, TexlStrings.IsMatchArg2, TexlStrings.IsMatchArg3 };
+        }
+
+        public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors)
+        {
+            var regexIsConstant = binding.IsConstant(args[1]);
+            var regexNodeKind = argTypes[1].Kind == DKind.OptionSetValue ? argTypes[1].OptionSetInfo.BackingKind : argTypes[1].Kind;
+
+            if (!regexIsConstant || regexNodeKind != DKind.String)
```

Comment:
>!regexIsConstant ||  [](http://example.com/codeflow?start=16&length=20)

From our internal discussion, we don't need to check for constant regex in this function anymore